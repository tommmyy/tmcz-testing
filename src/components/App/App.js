import React, { Component, Fragment } from 'react';
import { ThemeProvider, createGlobalStyle } from 'styled-components';
import { Button, Box } from 'rebass';
import theme from './theme';

const GlobalStyle = createGlobalStyle({
	body: {
		margin: 0,
		padding: 0,
	},
	input: { padding: '8px' },
});

class App extends Component {
	state = { age: '', clicked: false };

	handleAgeChange = (e) => this.setState({ age: e.target.value });
	handleClick = (e) => this.setState({ clicked: true });

	render() {
		const { age, clicked } = this.state;

		return (
			<ThemeProvider theme={theme}>
				<Fragment>
					<GlobalStyle />
					<p className="Age">{age}</p>
					<p className="Clicked">{clicked ? 'Clicked!' : ''}</p>
					<Box>
						<input name="age" value={age} onChange={this.handleAgeChange} />
					</Box>
					<Button active onClick={this.handleClick} variant="primary">
						Click
					</Button>
				</Fragment>
			</ThemeProvider>
		);
	}
}

export default App;
