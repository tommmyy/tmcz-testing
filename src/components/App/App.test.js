import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
	it('matches snapshots', () => {
		const wrapper = shallow(<App />);

		expect(wrapper).toMatchSnapshot();
	});
});
