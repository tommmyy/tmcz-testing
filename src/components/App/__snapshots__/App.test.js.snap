// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshots 1`] = `
<ThemeProvider
  theme={
    Object {
      "Heading": Object {
        "lineHeight": 1.25,
      },
      "Text": Object {
        "fontFamily": "system-ui, sans-serif",
      },
      "colors": Object {
        "blue": "#07c",
        "lightgray": "#f6f6ff",
      },
      "fontSizes": Array [
        12,
        14,
        16,
        20,
        24,
        32,
        48,
        64,
      ],
      "fonts": Object {
        "mono": "Menlo, monospace",
        "sans": "system-ui, sans-serif",
      },
      "shadows": Object {
        "large": "0 0 24px rgba(0, 0, 0, .125)",
        "small": "0 0 4px rgba(0, 0, 0, .125)",
      },
      "space": Array [
        0,
        4,
        8,
        16,
        32,
        64,
        128,
        256,
      ],
    }
  }
>
  <GlobalStyleComponent />
  <div
    className="App"
  >
    <section
      className="Section"
    >
      <p>
        Hello world
      </p>
      <ul>
        <li>
          first
        </li>
        <li>
          second
        </li>
      </ul>
      <input
        name="age"
      />
      <Styled(styled.div)
        active={true}
        as="button"
        bg="blue"
        border={0}
        borderRadius={4}
        color="white"
        fontSize="inherit"
        fontWeight="bold"
        m={0}
        px={3}
        py={2}
        variant="primary"
      >
        Click
      </Styled(styled.div)>
    </section>
  </div>
</ThemeProvider>
`;
